File: backtracking/subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        ans.add(new ArrayList<>());

        for(int ele: nums){
            int len = ans.size();
            for(int i = 0; i<len; i++){
                ArrayList<Integer> temp = new ArrayList<>(ans.get(i));
                temp.add(ele);
                ans.add(temp);
            }
        }
        return ans;
    }
}

File: backtracking/subsets.py
class Solution(object):
    def subsets(self, nums):
        answer = []
        def backtrack(curr,i):
            if i == len(nums):
                answer.append(curr[:])
                return

            curr.append(nums[i])
            backtrack(curr,i+1)
             
            curr.pop()

            backtrack(curr, i+1)

        backtrack([],0)
        return answer